src/hooks/useMainProcess.ts

```diff
@@ -67,7 +68,15 @@ export const useMainProcess = () => {
       if (state?.phase === 'success') {
         trackEvent(createTransactionEvent('transaction_success', state));
       } else if (state?.failure !== undefined) {
-        trackEvent(createTransactionEvent('transaction_failure', state));
+        const currentPhase = state?.phase;
+        const currentPhaseIndex = Object.keys(OFFRAMPING_PHASE_SECONDS).indexOf(currentPhase);
+
+        trackEvent({
+          ...createTransactionEvent('transaction_failure', state),
+          event: 'transaction_failure',
```

Comment:
Tiny nit: this `event` here should not be required.

Reply:
True but if I don't include it, TypeScript will complain
```
TS2353: Object literal may only specify known properties, and phase_name does not exist in type TransactionEvent
```
for `phase_name` and `phase_index`. I can either add a cast to `as TransactionFailureEvent` for the whole object or keep the `event: 'transaction_failure'` as is.